### DISCUSSION TITLE
AI Stuffs - Third-party integration & Vector DB Support

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#3213] [General] [answered]
Hi Everyone,

Is there any plans for “AI Stuffs - Third-party integration & Vector DB Support”?

What is the recommended way to integrate Third-party AI applications/services integration in Oqtane. (Let’s assume, we wanted to integrate Semantic Kernel/ChatGPT/Google Gemini and https://coframe.ai - how do I do that?)

AI Vector Databases such as https://milvus.io, https://weaviate.io , etc., can be compatible in the nearest future? If so it would be great and good for Oqtane’s future..

**Comment by sbwalker** [comment]
@Sharafudeen have you seen these repos:

https://github.com/oqtane/Oqtane.CoAdminPOC
https://github.com/oqtane/OqtaneOpenAIConfigurator

This work was done by @ADefWebserver a few months ago.

**Reply by Sharafudeen** [reply]
The links are not working.

**Reply by sbwalker** [reply, marked as answer]
Ahh... the links may be private repos. They were POCs for specific concepts for Oqtane... namely a "co-pilot" which could assist with building a site. @ADefWebserver was able to create a module where you could speak specific commands such as "add page" or "add module" and it would construct JSON which could be executed.  It was functional... although in reality it was still quicker to just use the standard Oqtane user interface to accomplish the tasks. And I think this is where the challenge with AI resides... determining the specific use cases where it can really add value. The other challenge is the technical landscape for AI is rapidly changing so it seems risky to put too much effort into a major integration as you will end up rewriting it in 3 months. So we will not be integrating any specific AI technology into the core framework at this time.... but  there is absolutely nothing stopping you from creating your own AI based solutions on Oqtane today.

**Comment by ADefWebserver** [comment]
For samples specifically using vector databases see: [What Are Embeddings and Vectors (And Why Should I Care?)](https://blazorhelpwebsite.com/ViewBlogPost/12068). Also see: [Use a Poor Developers Vector Database to Implement The RAG Pattern](https://blazorhelpwebsite.com/ViewBlogPost/5066) this specifically uses SQL server to store vector data. This is wayyy  less expensive than implementing a 3rd party vector database provider.

**Comment by ADefWebserver** [comment]
@sbwalker @Sharafudeen I made the repositories public
